USING_ELEKTRA.md

```diff
@@ -0,0 +1,136 @@
+ This guarantees that certain mistakes cannot occur like setting the background color to *greeen* which could
+ potentially crash the application. If you have followed the specification installation process correctly
+ from the [INSTALL.md](INSTALL.md), you now have the full specification [mounted](https://www.libelektra.org/tutorials/mount-configuration-files)
+ under the `spec` namespace. 
+ 
+ The specification for `.../curses/background` background for example
+ tells us that only `red, black, green, yellow, blue, magenta, cyan, white` are valid and that the value given 
+ has to be of type string. Elektra will prevent you to set invalid values which will help reduce errors and misconfiguration.
+  All these data and much more (eg. default values, descriptions, etc.) are saved as 
+ [metadata](https://www.libelektra.org/devdocu/metadata)
+ for each and every key under the `spec` namespace.
+ 
+ The following sections will teach you on how to inspect all relevant settings
+  and specifications and change desired configuration settings accordingly.
+ There best way to change settings is done via `kdb set`
+ since it integrates the best with specifications for configurations.
+ 
+### Changing Configurations
+ 
+Using the native command line is the best possibility to find out which configuration settings are available.
+If you want to see all available configuration settings for the `server`, you can simply call
+```sh
+kdb ls '/sw/lcdproc/lcdd/#0/current/server'
+#> /sw/lcdproc/lcdd/#0/current/server/autorotate
+#> /sw/lcdproc/lcdd/#0/current/server/backlight
+#> /sw/lcdproc/lcdd/#0/current/server/bind
+#> /sw/lcdproc/lcdd/#0/current/server/driver
+...
+```
+To see all available configuration settings at all you can execute `kdb ls '/sw/lcdproc/lcdd/#0/current'`.
+To see the current value you can simply call `kdb get` on the respective key.
+Now lets take the curses driver for example again and lets see what is the 
+current background color is:
+```sh
+kdb get '/sw/lcdproc/lcdd/#0/current/curses/background'
+#> cyan
+```
+The result comes from the set `default` metadata which is saved for the key. 
+You can see all metadata by calling `kdb lsmeta`:
+```sh
+kdb lsmeta '/sw/lcdproc/lcdd/#0/current/curses/background'
+#> check/enum/#0
+#> check/enum/#1
+#> check/enum/#2
+#> check/enum/#3
+#> check/enum/#4
+#> check/enum/#5
+#> check/enum/#6
+#> check/enum/#7
+#> check/type
+#> default
+#> description
+#> type
+```
+Here we can als see `check/enum/#0-7` which is used by Elektra to assert
+for correct configuration settings. You can query for the values by calling
+`kdb getmeta`
+```sh
+kdb getmeta '/sw/lcdproc/lcdd/#0/current/curses/background' 'check/enum/#0'
+#> red
+```
+This `.../#<number>` notation is the typical array notation which is used by Elektra.
+If you want to query all metadata and see their values you can use this little script:
+```sh
+kdb lsmeta '/sw/lcdproc/lcdd/#0/current/curses/background' \
```

Comment:
We should put this into a command and not encourage people to copy&paste it.

Reply:
As soon as https://github.com/ElektraInitiative/libelektra/issues/2412 is finished this can be changed

Reply:
Can you please create a PR to fix it?

Reply:
Still open.

Reply:
Unfortunately this is out of scope for me at the moment. I might try to code the terminal command at a later point in time.

Reply:
This is a copy&paste of one line into a script?

Reply:
Do you have a similar PR where this is done so I can see what exactly is done? I doubt that it just copy&paste but also some additional checks and documentation on how to use the command line tool

Reply:
`git log -p scripts` show you how previous scripts were added.